### PR TITLE
Fix: rewrite message typing and construct

### PR DESCRIPTION
**Pull request by yanyongyu**
Remove mapping construct for class `Message` and improve typing for `Message` and `MessageSegment`

### CODE REVIEW

File: nonebot/adapters/_message.py [hunk 1]
```diff
@@ -101,18 +102,18 @@ class Message(List[TMS], abc.ABC):
...
+        super().__init__()
         if message is None:
             return
-        elif isinstance(message, Message):
-            self.extend(message)
+        elif isinstance(message, str):
+            self.extend(self._construct(message))
         elif isinstance(message, MessageSegment):
             self.append(message)
+        elif isinstance(message, Iterable):
```
Comment (j1g5awi) [inline nonebot/adapters/_message.py:115]:
```suggestion
        elif isinstance(message, Iterable[TMS]):
```

File: nonebot/adapters/_message.py [hunk 2]
```diff
@@ -163,51 +164,93 @@ def __get_validators__(cls):
...
 
-    def __iadd__(self: TM, other: Union[str, Mapping, Iterable[Mapping]]) -> TM:
-        if isinstance(other, MessageSegment):
+    def __iadd__(self: TM, other: Union[str, TMS, Iterable[TMS]]) -> TM:
+        if isinstance(other, str):
+            self.extend(self._construct(other))
+        elif isinstance(other, MessageSegment):
             self.append(other)
-        elif isinstance(other, Message):
+        elif isinstance(other, Iterable):
```
Comment (j1g5awi) [inline nonebot/adapters/_message.py:199]:
```suggestion
        elif isinstance(other, Iterable[TMS]):
```

**Review by j1g5awi — CHANGES_REQUESTED** [review]

File: nonebot/adapters/_message.py [hunk 3]
```diff
@@ -167,47 +154,79 @@ def _validate(cls, value):
 
     @staticmethod
     @abc.abstractmethod
-    def _construct(msg: Union[str, Mapping, Iterable[Mapping], Any]) -> Iterable[TMS]:
+    def _construct(msg: Union[str, Any]) -> Iterable[TMS]:
```
[the remark this answers is not in the document]
Comment (yanyongyu) [reply nonebot/adapters/_message.py]: 因为TMS不会进_construct了，在init里处理掉了

File: nonebot/adapters/_message.py (hunk 1 above)
Comment (yanyongyu) [reply nonebot/adapters/_message.py:115]: Iterable[TMS]不支持isinstance

File: nonebot/adapters/_message.py (hunk 2 above)
Comment (yanyongyu) [reply nonebot/adapters/_message.py:199]: Iterable[TMS]不支持isinstance

**Pull request merged**